### ISSUE TITLE
Error: type mismatch: got <seq[uint8]> but expected 'seq[ColorId]'

### ISSUE THREAD

**Issue by narimiran** [ftsf/nico#11]
When I try to run your tests, this error pops up in various places.

Tested with Nim 0.19.0, 0.19.4 and 0.19.9

**Comment by kidandcat** [comment]
Same here with nim 0.20

**Comment by dexelon** [comment]
It's fixed in the `nim0.20.0` branch.

**Comment by UNIcodeX** [comment]
not for me. `choosenim 0.20.0` and then `nim c -r platformer.nim` gives the following error:

`~\.nimble\pkgs\nico-0.1.0\nico\backends\common.nim(323, 30) Error: type mismatch: got <ColorId> but expected 'uint8'`

**Comment by narimiran** [comment]
> It's fixed in the nim0.20.0 branch.

Since we're currently on Nim 1.2, and this fix which is in that branch will (probably) make 0.19.x (released on 26 September 2018!), 0.20.x, 1.0.x, and 1.2.x work, and [according to our survey](https://nim-lang.org/blog/2020/02/18/community-survey-results-2019.html#nim-versions), almost nobody uses 0.18.x or lower — how about merging in into master branch?

**Comment by ftsf** [comment]
Hi sorry about the delays, I've got a working branch for Nim 1.x that i'll need to push, there are some breaking changes though. I'll try and get it up this week.

**Comment by krisppurg** [comment]
@UNIcodeX I believe you have been misunderstood; what I mean by that is that the github branch of nico (`nim0.20.0`) fixes the issue.

I've tested the `master` nico branch in nim 0.20.0 that doesn't work, of course.

First thing what you got to do is, install the `nim0.20.0` nico branch via git or nimble, go to the `.nimble/pkgs` directory, drag the folder (fixed `nim0.20.0` branch one), now first you have the new version of nico (`nim0.20.0` branch version), right? Delete the original pkg folder (`nico-1.0`), rename the new folder as the original pkg folder of nico, so nim thinks its the nico that we want and there you go.

Now, I know this is a late reply and I am aware of the new nico update releasing this week, so there isn't that much of a point of that solution.

**Comment by UNIcodeX** [comment]
@krisppurg You are correct. Initially I had misunderstood, but then re-read your reply and realized you you were saying that there is a branch named nim0.20.0 for nico. I did try to checkout that branch and installed it via `nimble develop` within the cloned repo, but that still gave me an error. This time with the sdl2 wrapper. I figured I'd just wait at this point since @ftsf said he'd push an update this week. 

Thank you for the clarification, as I may not have realized otherwise.
